table/tables/state_remote.go

```diff
@@ -155,7 +155,7 @@ func (h *stateRemoteHandle) LockForWrite(ctx context.Context, tid int64, leaseDu
 }
 
 func (h *stateRemoteHandle) lockForWriteOnce(ctx context.Context, tid int64, leaseDuration time.Duration) (waitAndRetry time.Duration, ts uint64, err error) {
-	err = h.runInTxn(ctx, func(ctx context.Context, now uint64) error {
+	err = h.runInTxn(ctx, true, func(ctx context.Context, now uint64) error {
```

Comment:
LockForWrite use pessimistic lock (must success for writing)
LockForRead use optimistic lock and ignore txn conflict error

Because if LockForWrite use optimistic and meet a txn conflict, the error will be throw to the user directly.

Reply:
When using a pessmistic txn. Should we use `select for update` instead of `select`

Reply:
OK, I'll just fix the panic and leave other changes later~